rtsp.go

```diff
@@ -232,8 +232,18 @@ func (rc *rtspCamera) Close(_ context.Context) error {
 	rc.unsubscribeAll()
 	rc.activeBackgroundWorkers.Wait()
 	rc.closeConnection()
-	rc.avFramePool.close()
 	rc.mimeHandler.close()
+	// Clean up latestFrame cache if it exists. This is necessary to ensure that the frame is properly
+	// freed when the avFramePool is closed.
+	rc.latestFrameMu.Lock()
```

Comment:
one q - we're already in a lock; closeMu it is a RWMutex, but latestFrameMu is a Mutex. Why was the extra lock necessary to get the latest frame.

Do we need to free the frame cache at other parts of the code?

Reply:
Good question! I thought about removing this as well but concerned about the following:
`Close` is triggered and we  `closeConnection` which cancels the context on `gortsplib.Client. It seems like we should be safe to not have to lock since we will no longer have `handleLatestFrame` hits downstream of the `c.client.OnPacketRTP` callback. But, there still could be a race condition where we close the client and move on to the rest of Close but it takes some time for the client to go silent.

The only other concern is conflicts with the `Image` method but it has a `closeMu` read lock on it so seem to be safe there.

Reply:
Okay - keep an eye on this behaviour as you battle test.